analysis/filter/StructureModifier/src/test/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/NodeRemoverTest.kt

```diff
@@ -0,0 +1,103 @@
+        sampleProject = ProjectDeserializer.deserializeProject(bufferedReader)
+    }
+
+    @Test
+    fun `Non existent path leads to replication of the project`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/somethig"))
+
+        Assertions.assertThat(result).isEqualToComparingFieldByFieldRecursively(sampleProject)
+    }
+
+    @Test
+    fun `Non affected nodes are kept`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/src/main"))
+
+        val testFolder = result.rootNode.children.first().children.first()
+        Assertions.assertThat(testFolder.name).isEqualTo("test")
+    }
+
+    @Test
+    fun `specified nodes are removed`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/src/main"))
+
+        val srcContent = result.rootNode.children.first().children
+        Assertions.assertThat(srcContent.size).isEqualTo(2)
+        Assertions.assertThat(srcContent.map { it.name }).doesNotContain("main")
+    }
+
+    @Test
+    fun `multiple nodes are removed`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/src/main/file1.java", "root/src/folder3/"))
+
+        val srcContent = result.rootNode.children.first().children
+        val mainContent = srcContent.first().children
+        Assertions.assertThat(srcContent.map { it.name }).doesNotContain("folder3")
+        Assertions.assertThat(mainContent.map { it.name }).containsOnly("file2.java")
+    }
+
+    @Test
+    fun `project name is set to specified name`() {
+        val subProjectExtractor = NodeRemover(sampleProject, "fooBar")
+
+        val result = subProjectExtractor.remove(arrayOf())
+
+        Assertions.assertThat(result.projectName).isEqualTo("fooBar")
+    }
+
+    @Test
+    fun `Affected edges are removed`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+
+        val result = subProjectExtractor.remove(arrayOf("/root/foo"))
+
+        Assertions.assertThat(result.edges.size).isEqualTo(1)
+        Assertions.assertThat(result.edges.first()).isEqualToComparingFieldByFieldRecursively(sampleProject.edges.last())
+        Assertions.assertThat(result.edges.map { it.fromNodeName }).doesNotContain("/roo/foo")
+        Assertions.assertThat(result.edges.map { it.toNodeName }).doesNotContain("/roo/foo")
+    }
+
+    @Test
+    fun `Affected edges are removed for multiple removals`() {
+        val subProjectExtractor = NodeRemover(sampleProject, null)
+        val toExclude = arrayOf("/root/foo/file1", "root/else/")
+
+        val result = subProjectExtractor.remove(toExclude)
+
+        val shouldBeExcluded = listOf("/root/foo/file1", "root/else/file1")
```

Comment:
I think testing against `listOf("/root/foo/file1", "root/else")`makes more sense, as the test removes exactly those two folders